### PR TITLE
test: Add SPIFFE exchange, malformed bearer, and credential error tests

### PR DESCRIPTION
**Pull request by huang195**
## Summary

Cover the highest-value remaining gaps in go-processor token exchange:

**SPIFFE federated exchange (3 subtests):**
- Succeeds with JWT-SVID file, verifies client_assertion_type and client_assertion params
- Fails gracefully when SVID file is missing (no Keycloak call made)
- Fails gracefully when SVID file is empty

**Outbound edge cases (3 tests):**
- Malformed (non-Bearer) Authorization header on routed host returns 503
- clientCredentialsGrant surfaces Keycloak 401 errors correctly
- client_credentials fallback failure on routed host returns 503 (not passthrough)

**Coverage changes (this branch only):**
- exchangeToken: 70.8% -> 87.5%
- handleOutbound: 87.7% -> 93.8%

Combined with #304 (inbound) and #305 (config/utility), total go-processor coverage reaches ~65-70%.

## Test plan

- [x] All 31 tests pass locally with -race flag
- [ ] CI passes (requires #303 to run tests in CI)

### CODE REVIEW

File: authbridge/authproxy/go-processor/main_test.go [hunk 1]
```diff
@@ -437,6 +437,177 @@ func TestDefaultBypassPaths(t *testing.T) {
...
+		if err == nil {
+			t.Fatal("expected error when JWT-SVID file is empty")
+		}
+	})
+}
+
+// TestOutboundMalformedBearerToken verifies that an outbound request with an
+// Authorization header that doesn't have the "Bearer " prefix gets denied (503).
+func TestOutboundMalformedBearerToken(t *testing.T) {
+	saved := saveGlobals()
```
Comment (pdettori) [inline authbridge/authproxy/go-processor/main_test.go:1079]: nit: Unlike the "svid file missing" subtest (which checks `strings.Contains(err.Error(), "JWT-SVID unavailable")`), this subtest only checks `err != nil`. Consider adding a similar error message assertion for consistency — it guards against the wrong error path being taken (e.g. a network error from the mock server instead of the expected empty-file validation).

File: authbridge/authproxy/go-processor/main_test.go [hunk 2]
```diff
@@ -437,6 +437,177 @@ func TestDefaultBypassPaths(t *testing.T) {
...
+		// Write a fake JWT-SVID to a temp file
+		dir := t.TempDir()
+		svidPath := filepath.Join(dir, "jwt_svid.token")
+		os.WriteFile(svidPath, []byte("fake-jwt-svid-content"), 0644)
+
+		token, err := exchangeToken("spiffe://localtest.me/ns/team1/sa/agent", "", server.URL,
+			"subject-token", "auth-target", "openid auth-target-aud",
+			true, svidPath, "")
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
```
Comment (pdettori) [inline authbridge/authproxy/go-processor/main_test.go:1016]: nit (cosmetic): `os.WriteFile` return value is unchecked here and in the empty-SVID subtest. Idiomatic for Go tests since the test would fail downstream, but a quick `if err != nil { t.Fatal(err) }` would make setup failures immediately obvious.

**Review by pdettori — APPROVED** [review]
Clean test-only PR — well-structured tests with proper resource cleanup, global state isolation, and good assertion coverage. Two minor nits below, neither blocking.

**Areas reviewed**: Go tests, commit conventions, security
**CI**: all passing
